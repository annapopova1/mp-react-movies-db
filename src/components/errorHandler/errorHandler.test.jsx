import React from 'react';
import ErrorHandler from './errorHandler';

describe('<ErrorHandler/>', () => {
  const ComponentWithError = () => { throw new Error(); };

  test('should render child elements of ErrorHandler component', () => {
    const errorHandler = shallow(<ErrorHandler><ComponentWithError /></ErrorHandler>);
    expect(errorHandler).toMatchSnapshot();
  });

  test('should render error content for cought exceptions', () => {
    const errorHandler = shallow(<ErrorHandler><ComponentWithError /></ErrorHandler>);
    errorHandler.setState({ hasError: true });
    expect(errorHandler).toMatchSnapshot();
  });
});
