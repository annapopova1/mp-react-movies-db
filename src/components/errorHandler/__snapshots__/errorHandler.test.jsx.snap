// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorHandler/> should render child elements of ErrorHandler component 1`] = `<ComponentWithError />`;

exports[`<ErrorHandler/> should render error content for cought exceptions 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  The application was breaking, try again later!
</div>
`;
